assign2/src/model/filter/types.ts

```diff
@@ -0,0 +1,37 @@
+import Tag from "../objects/Tag";
+
+
+export enum QuestionFilter{
+    ALL_POSTS = "ALL_POSTS",
+    FILTER_BY_TITLE = "FILTER_BY_TITLE",
+    FILTER_BY_TAG = "FILTER_BY_TAG",
+}
+
+export const SET_FILTER = "SET_FILTER";
+export const SET_SEARCHED_TITLE = "SET_SEARCHED_TITLE";
+export const SET_SEARCHED_TAG = "SET_SEARCHED_TAG";
```

Comment:
Again, just as an FYI, you could also define more human readable values for the action types (https://blog.strongbrew.io/Redux-best-practices/#actiontypes). This is especially helpful if you use something like the Redux chrome plugin to debug.